api/client/alpn_websocket.go

```diff
@@ -124,8 +124,8 @@ func (c *websocketALPNClientConn) readLocked(b []byte) (int, error) {
 			return 0, io.EOF
 		case ws.OpPing:
 			pong := ws.NewPongFrame(frame.Payload)
-			if err := c.writeFrame(pong); err != nil {
-				return 0, trace.Wrap(err)
+			if err := c.writeFrameLocked(pong); err != nil {
+				return 0, err
```

Comment:
Is there a reason we're no longer wrapping these errors? If upstream code requires that they're not wrapped, maybe a comment here would be justified.

Reply:
the write here should be locked otherwise there is a race with writing data packets? was it deadlocking here without this change?